neutron/templates/configmaps/metadata-agent.ini.yaml.j2

```diff
@@ -3,6 +3,30 @@ kind: ConfigMap
 data:
   metadata-agent.ini: |+
     [DEFAULT]
-    nova_metadata_ip = nova-api
+    debug = {{ misc.debug }}
+
+    # Neutron credentials for API access
+    auth_plugin = password
+    auth_url = {{ keystone.auth_url }}
+    auth_uri = {{ keystone.auth_uri }}
+    auth_region = {{ keystone.neutron_region_name }}
+    project_domain_id = default
+    user_domain_id = default
+    project_name = service
+    username = {{ keystone.admin_user }}
+    password = {{ keystone.admin_password }}
+    endpoint_type = adminURL
+
+    # Nova metadata service IP and port
+    nova_metadata_ip = {{ network.ip_address }}
     nova_metadata_port = {{ metadata.port }}
+
+    # Metadata proxy shared secret
     metadata_proxy_shared_secret = {{ metadata.secret }}
```

Comment:
Rename to `neutron.metadata_secret`, add to manifest with default and be sure to duplicate the variable (and its default) into the nova's manifest as well and use it in nova.conf.